src/installer/installer_test.go

```diff
@@ -1,16 +1,166 @@
 package installer
 
-//import (
-//	"github.com/eranco74/assisted-installer/src/config"
-//	"github.com/sirupsen/logrus"
-//)
-//
-//var _ = Describe("installer", func() {
-//	var conf = config.Config{Role: "master", ClusterID: "cluster-id", Device: "/dev/vda", S3EndpointURL: "https://s3.com", S3Bucket: "test"}
-//	var l = logrus.New()
-//	i := NewAssistedInstaller(l, conf)
-//
-//	It("master role happy flow", func() {
-//		i.InstallNode()
-//	})
-//})
+import (
+	"fmt"
+	"github.com/eranco74/assisted-installer/src/config"
+	"github.com/eranco74/assisted-installer/src/ops"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+)
+
+func TestValidator(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "installer_test")
+}
+
+var _ = Describe("installer master role", func() {
+	var (
+		l            = logrus.New()
+		mockops      *ops.MockOps
+		i            *installer
+		bootstrapIgn = "bootstrap.ign"
+		masterIgn    = "master.ign"
+	)
+	device := "/dev/vda"
+	baseUrl := "https://s3.com/test/cluster-id/"
+	l.SetOutput(ioutil.Discard)
+	mkdirSuccess := func() {
+		mockops.EXPECT().Mkdir(installDir).Return(nil).Times(1)
+	}
+	curlSuccess := func(fileName string) {
+		mockops.EXPECT().ExecPrivilegeCommand("curl", "-s", baseUrl+fileName, "-o", filepath.Join(installDir, fileName)).Return("", nil).Times(1)
+	}
+	writeToDiskSuccess := func() {
+		mockops.EXPECT().WriteImageToDisk(filepath.Join(installDir, "master.ign"), device).Return(nil).Times(1)
+	}
+	rebootSuccess := func() {
+		mockops.EXPECT().Reboot().Return(nil).Times(1)
+	}
+	Context("Bootstrap role", func() {
+		conf := config.Config{Role: "bootstrap", ClusterID: "cluster-id", Device: "/dev/vda", S3EndpointURL: "https://s3.com", S3Bucket: "test"}
+		BeforeEach(func() {
+			ctrl := gomock.NewController(GinkgoT())
```

Comment:
suggesting to create .the controller in a `BeforeEach` under `Describe` and have `AfterEach` with `ctrl.Finish()`